auth/authenticator.go

```diff
@@ -97,7 +97,7 @@ func (l *LsatAuthenticator) FreshChallengeHeader(r *http.Request,
 
 	str := fmt.Sprintf("LSAT macaroon=\"%s\", invoice=\"%s\"",
 		base64.StdEncoding.EncodeToString(macBytes), paymentRequest)
-	header := r.Header
```

Comment:
Why start from the blank header vs inheriting it from the request?

Reply:
It didn't really make sense to start with the request header fields in the first place. Since the only (common) fields that are both valid in the request and response are the `Content-Type` and `Content-Length`. And the second one was what caused the issue.
So this was just wrong and we haven't noticed before.

Reply:
Gotcha, can we add a `POST` case to this test case so we're protected against regressions: https://github.com/lightninglabs/aperture/blob/master/proxy/proxy_test.go#L89